applications/StructuralMechanicsApplication/custom_processes/postprocess_eigenvalues_process.cpp

```diff
@@ -145,28 +145,17 @@ struct VtkEigenOutputWrapper : public BaseEigenOutputWrapper
 
 } // helpers namespace
 
-PostprocessEigenvaluesProcess::PostprocessEigenvaluesProcess(ModelPart& rModelPart,
-                                                                Parameters OutputParameters)
-                                                                : mrModelPart(rModelPart),
-                                                                mOutputParameters(OutputParameters)
+PostprocessEigenvaluesProcess::PostprocessEigenvaluesProcess(
+    Model& rModel,
+    Parameters OutputParameters)
+    : mOutputParameters(OutputParameters)
 {
-    Parameters default_parameters(R"(
-        {
-            "result_file_name"              : "Structure",
-            "file_format"                   : "vtk",
-            "file_label"                    : "step",
-            "result_file_format_use_ascii"  : false,
-            "folder_name"                   : "EigenResults",
-            "save_output_files_in_folder"   : true,
-            "animation_steps"               : 20,
-            "label_type"                    : "frequency",
-            "list_of_result_variables"      : ["DISPLACEMENT"]
-        }  )"
-    );
-
-    mOutputParameters.RecursivelyValidateAndAssignDefaults(default_parameters);
-
+    mOutputParameters.RecursivelyValidateAndAssignDefaults(GetDefaultParameters());
+    mpModelPart = &(rModel.GetModelPart(mOutputParameters["model_part_name"].GetString()));
     const std::string folder_name = mOutputParameters["folder_name"].GetString();
+    if (mOutputParameters["wipe_results_folder"].GetBool()) {
+        std::filesystem::remove_all(folder_name);
+    }
     if (!std::filesystem::exists(folder_name)) {
         std::filesystem::create_directories(folder_name);
     }
```

Comment:
```suggestion
    if (OutputParameters["save_output_files_in_folder"].GetBool()) {
        if (mOutputParameters["wipe_results_folder"].GetBool()) {
            std::filesystem::remove_all(folder_name);
        }
        if (!std::filesystem::exists(folder_name)) {
            std::filesystem::create_directories(folder_name);
        }
    }
```

final comment I promise :D

Reply:
Makes perfect sense :+1: